### PR TITLE
Feature/2844/tooltip design system

### LINKED ISSUE #2844: (SP: 3) Create Tooltip component aligned with design system
#### Description:
Implement a reusable `Tooltip` component that meets the design system standards. This component will be used throughout the application to maintain consistency in tooltip styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=21836-151550&t=FRlxrBiejnE3M9Qb-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  - Support different variants (large, small, outlined).
  - Ensure hover, focus, active, and disabled states are defined according to the design system.
  - Provide a title for Tooltip.
  
- **Props:**
  - `variant`: Defines type (icon, text, icon & text).
  - `title `: Provide a title for the Tooltip.
  - `description`: Provide a description for the Tooltip.
  - `position`: Supports different positions (up, down, right, left, none).

Don't forget to cover component with tests and add to the storybook

### PR DESCRIPTION
**Pull request by markgol777**
image

Tooltip design system component created, covered with tests and added a storybook.

### CODE REVIEW

**Review by dudchakk — DISMISSED** [review]
check paddings, especially for icon only looltip, looks like your side paddings are smaller than the ones in the figma mockup)

**Comment by markgol777** [comment]
> check paddings, especially for icon only looltip, looks like your side paddings are smaller than the ones in the figma mockup)

Fixed the issue, about to commit
Screenshot 2024-11-25 at 22 45 26

File: src/design-system/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+const ToolTip: React.FC<TooltipProps> = ({
+  variant,
+  title,
+  description,
+  position
+}) => {
+  const textContent = (
+    <Box className={cn('s2s-tooltip-text-container')}>
+      <Typography className={cn('s2s-tooltip-title')}>{title}</Typography>
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/tooltip/Tooltip.tsx]: Remove redundant `cn` calls

File: src/design-system/components/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      {description && <Typography>{description}</Typography>}
+    </Box>
+  )
+
+  const iconContent = <DoneIcon className={cn('s2s-tooltip-icon')} />
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/tooltip/Tooltip.tsx]:
```suggestion
  const iconContent = <DoneIcon className='s2s-tooltip-icon' />
```

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/design-system/components/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  position
+}) => {
+  const textContent = (
+    <Box className={'s2s-tooltip-text-container'}>
+      <Typography className={'s2s-tooltip-title'}>{title}</Typography>
+      {description && <Typography>{description}</Typography>}
+    </Box>
+  )
+
+  const iconContent = <DoneIcon className={'s2s-tooltip-icon'} />
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/tooltip/Tooltip.tsx]: Curly braces are unnecessary here, so please remove them.
```suggestion
    <Box className='s2s-tooltip-text-container'>
      <Typography className='s2s-tooltip-title'>{title}</Typography>
      {description && <Typography>{description}</Typography>}
    </Box>
  )

  const iconContent = <DoneIcon className='s2s-tooltip-icon' />
```

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/design-system/components/tooltip/Tooltip.tsx (hunk 3 above)
Comment (markgol777) [reply src/design-system/components/tooltip/Tooltip.tsx]: yeah for sure I didn't notice

**Comment by Mav-Ivan** [comment]
add an icon as a prop so we can show icons if there's need

**Review by Renatavl — APPROVED** [review]

**Review by ShadowOfTheSpace — APPROVED** [review]

**Comment by markgol777** [comment]
> add an icon as a prop so we can show icons if there's need

Fixed

**Review by amoutens — APPROVED** [review]

**Pull request merged**